src/connector/src/source/cdc/mod.rs

```diff
@@ -15,36 +15,46 @@
 pub mod enumerator;
 pub mod source;
 pub mod split;
-
 use std::collections::HashMap;
+use std::marker::PhantomData;
 
-use anyhow::anyhow;
 pub use enumerator::*;
+use paste::paste;
 use risingwave_common::catalog::{ColumnDesc, Field, Schema};
-use risingwave_pb::connector_service::{SourceType, TableSchema};
-use serde::Deserialize;
+use risingwave_pb::connector_service::{PbSourceType, PbTableSchema, SourceType, TableSchema};
 pub use source::*;
 pub use split::*;
 
-pub const MYSQL_CDC_CONNECTOR: &str = "mysql-cdc";
-pub const POSTGRES_CDC_CONNECTOR: &str = "postgres-cdc";
-pub const CITUS_CDC_CONNECTOR: &str = "citus-cdc";
+use crate::impl_cdc_source_type;
+use crate::source::ConnectorProperties;
+
+pub const CDC_CONNECTOR_NAME_SUFFIX: &str = "-cdc";
 
-#[derive(Clone, Debug, Deserialize, Default)]
-pub struct CdcProperties {
-    /// Type of the cdc source, e.g. mysql, postgres
-    pub source_type: String,
+pub const MYSQL_CDC_CONNECTOR: &str = Mysql::CDC_CONNECTOR_NAME;
+pub const POSTGRES_CDC_CONNECTOR: &str = Postgres::CDC_CONNECTOR_NAME;
+pub const CITUS_CDC_CONNECTOR: &str = Citus::CDC_CONNECTOR_NAME;
+
+pub trait CdcSourceTypeTrait: Send + Sync + Clone + 'static {
+    const CDC_CONNECTOR_NAME: &'static str;
+    fn source_type() -> CdcSourceType;
+}
+
+impl_cdc_source_type!({ Mysql, "mysql" }, { Postgres, "postgres" }, { Citus, "citus" });
```

Comment:
Hmm, the macro is only used once here. I don't think it can reduce many LOC and seems not worth to me.

Reply:
Though the macro is used only once, its input is a list of cdc source. Inside the macro, and we can leverage the ability of list expansion inside macro to generate the same code for all cdc sources. So the code in macro is actually reused by each cdc source.

For example, the following code inside the macro generates the same code for all cdc sources

```
macro_rules! impl_cdc_source_type {
    ($({$source_type:ident, $name:expr }),*) => {
        $(
            paste!{
                #[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
                pub struct $source_type;
                impl CdcSourceTypeTrait for $source_type {
                    const CDC_CONNECTOR_NAME: &'static str = concat!($name, "-cdc");
                    fn source_type() -> CdcSourceType {
                        CdcSourceType::$source_type
                    }
                }

                pub type [< $source_type DebeziumSplitEnumerator >] = DebeziumSplitEnumerator<$source_type>;
            }
        )*
    }
}
```